Sources/AsyncHTTPClient/HTTPHandler.swift

```diff
@@ -694,69 +704,24 @@ extension HTTPClient {
+                self._isCancelled = true
+                return self._taskDelegate
             }
-            channel?.triggerUserOutboundEvent(TaskCancelEvent(), promise: nil)
-        }
 
-        @discardableResult
-        func setConnection(_ connection: Connection) -> Connection {
-            return self.lock.withLock {
-                self.connection = connection
-                if self.cancelled {
-                    connection.channel.triggerUserOutboundEvent(TaskCancelEvent(), promise: nil)
-                }
-                return connection
-            }
+            taskDelegate?.cancel()
         }
 
         func succeed<Delegate: HTTPClientResponseDelegate>(promise: EventLoopPromise<Response>?,
                                                            with value: Response,
                                                            delegateType: Delegate.Type,
                                                            closing: Bool) {
-            self.releaseAssociatedConnection(delegateType: delegateType,
-                                             closing: closing).whenSuccess {
-                promise?.succeed(value)
-            }
+            promise?.succeed(value)
         }
 
         func fail<Delegate: HTTPClientResponseDelegate>(with error: Error,
                                                         delegateType: Delegate.Type) {
-            if let connection = self.connection {
-                self.releaseAssociatedConnection(delegateType: delegateType, closing: true)
-                    .whenSuccess {
-                        self.promise.fail(error)
-                        connection.channel.close(promise: nil)
-                    }
-            } else {
-                // this is used in tests where we don't want to bootstrap the whole connection pool
-                self.promise.fail(error)
-            }
-        }
-
-        func releaseAssociatedConnection<Delegate: HTTPClientResponseDelegate>(delegateType: Delegate.Type,
-                                                                               closing: Bool) -> EventLoopFuture<Void> {
-            if let connection = self.connection {
-                // remove read timeout handler
-                return connection.removeHandler(IdleStateHandler.self).flatMap {
-                    connection.removeHandler(TaskHandler<Delegate>.self)
-                }.map {
-                    connection.release(closing: closing, logger: self.logger)
-                }.flatMapError { error in
-                    fatalError("Couldn't remove taskHandler: \(error)")
-                }
-            } else {
-                // TODO: This seems only reached in some internal unit test
-                // Maybe there could be a better handling in the future to make
-                // it an error outside of testing contexts
-                return self.eventLoop.makeSucceededFuture(())
-            }
+            self.promise.fail(error)
```

Comment:
Those functions are not hit from the new implementation. However they are hit from the old implementation. Since we don't want to remove the old implementation in this pr, they should stay for now. We will be able to remove them in a follow up pr.